ibm/data_source_ibm_atracker_routes.go

```diff
@@ -0,0 +1,185 @@
+// Copyright IBM Corp. 2021 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package ibm
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/IBM/platform-services-go-sdk/atrackerv1"
+)
+
+func dataSourceIBMAtrackerRoutes() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMAtrackerRoutesRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The name of this route.",
+			},
+			"routes": &schema.Schema{
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "A list of route resources.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The uuid of this route resource.",
+						},
+						"name": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of this route.",
+						},
+						"crn": &schema.Schema{
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The crn of this route type resource.",
+						},
+						"version": &schema.Schema{
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "The version of this route.",
+						},
+						"receive_global_events": &schema.Schema{
+							Type:        schema.TypeBool,
+							Computed:    true,
+							Description: "Whether or not all global events should be forwarded to this region.",
+						},
+						"rules": &schema.Schema{
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The routing rules that will be evaluated in their order of the array.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"target_ids": &schema.Schema{
+										Type:        schema.TypeList,
+										Computed:    true,
+										Description: "The target ID List. Only one target id is supported.",
+										Elem: &schema.Schema{
+											Type: schema.TypeString,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceIBMAtrackerRoutesRead(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	atrackerClient, err := meta.(ClientSession).AtrackerV1()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	listRoutesOptions := &atrackerv1.ListRoutesOptions{}
+
+	routeList, response, err := atrackerClient.ListRoutesWithContext(context, listRoutesOptions)
+	if err != nil {
+		log.Printf("[DEBUG] ListRoutesWithContext failed %s\n%s", err, response)
+		return diag.FromErr(err)
```

Comment:
@mkistler Can we enhance the error to return the error as combination of both err and response...because response contains more details of failure than error?

Reply:
We can.  Can you point to an example in the current IBM Terraform provider that I can mimic?

Reply:
Like this:
```
return fmt.Errorf("[DEBUG] Create SSH Key %s\n%s", err, response)
```

Or "beautify_error" ...